### PR TITLE
ref(js): Marginally improve getCookie

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/utils/getCookie.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-        return decodeURIComponent(cookie.substring(name.length + 1));
-      }
-    }
+  if (!document.cookie || document.cookie === '') {
+    return null;
   }
-  return null;
+
+  const cookies = document.cookie.split(';');
+  const cookie = cookies.find(c => c.substring(0, name.length + 1) === `${name}=`);
```
Comment (priscilawebdev) [inline static/app/utils/getCookie.tsx:10]: shouldn't we still keep the `trim()`?

Comment (evanpurkhiser) [reply static/app/utils/getCookie.tsx:10]: oops, yep good catch!!

Comment (evanpurkhiser) [inline static/app/utils/getCookie.tsx:10]:
```suggestion
  const cookie = cookies.find(c => c.trim().substring(0, name.length + 1) === `${name}=`);
```

**Review by priscilawebdev — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Actually my trim fix wasn't totally right. Let me fix again

**Comment by JoshFerge** [comment]
why are we changing this at all?

**Comment by evanpurkhiser** [comment]
> why are we changing this at all?

I stumbled on this little utility while looking through the utils folder and thought it could be a bit more readable.

**Pull request merged**